dockerize.go

```diff
@@ -49,6 +66,59 @@ func (s *sliceVar) String() string {
 	return strings.Join(*s, ",")
 }
 
+func waitForDependencies() {
+	dependencyChan := make(chan struct{})
+
+	go func() {
+		for _, host := range waitFlag {
+			log.Println("Waiting for host:", host)
+			u, err := url.Parse(host)
+			if err != nil {
+				log.Fatalf("bad hostname provided: %s. %s", host, err.Error())
+			}
+
+			switch u.Scheme {
+			case "tcp", "tcp4", "tcp6":
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					for {
+						conn, _ := net.DialTimeout(u.Scheme, u.Host, waitTimeoutFlag)
+						if conn != nil {
+							log.Println("Connected to", u.String())
+							return
+						}
+					}
+				}()
+			case "http", "https":
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					for {
+						_, err := http.Get(u.String())
```

Comment:
Seems like this should check to make sure a `200` response is returned.  What if a `404` or `500` is returned by the dependency? 

You an also specify a timeout using a custom `http.Client` and/or `Transport`.